managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/OpFindNewest.java

```diff
@@ -115,34 +118,131 @@ public void readEntryComplete(Entry entry, Object ctx) {
             } else {
                 // start binary search
                 state = State.searching;
-                searchPosition = ledger.getPositionAfterN(startPosition, mid(), PositionBound.startExcluded);
+                this.mid = mid();
+                searchPosition = ledger.getPositionAfterN(startPosition, this.mid, PositionBound.startExcluded);
                 find();
             }
             break;
         case searching:
             if (condition.test(entry)) {
                 // mid - last
                 lastMatchedPosition = position;
-                min = mid();
+                min = mid;
             } else {
                 // start - mid
-                max = mid() - 1;
+                max = mid - 1;
             }
+            this.mid = mid();
 
             if (max <= min) {
                 callback.findEntryComplete(lastMatchedPosition, OpFindNewest.this.ctx);
                 return;
             }
-            searchPosition = ledger.getPositionAfterN(startPosition, mid(), PositionBound.startExcluded);
+            searchPosition = ledger.getPositionAfterN(startPosition, this.mid, PositionBound.startExcluded);
             find();
         }
     }
 
     @Override
     public void readEntryFailed(ManagedLedgerException exception, Object ctx) {
+        if (exception instanceof ManagedLedgerException.NonRecoverableLedgerException
+            && ledger.getConfig().isAutoSkipNonRecoverableData()) {
+            log.info("[{}] Ledger {} is not recoverable, skip non-recoverable data, state:{}", ledger.getName(),
```

Comment:
Do we need to add `try/catch` block to this new added logic in case of exceptions causing the callback failing to execute?

Reply:
Add catch